lib/router.js

```diff
@@ -97,9 +98,17 @@ Router.prototype._buildPath = function route(node, path, value) {
 /**
  * Load and initialize a module
  */
-Router.prototype._loadModule = function(modDef, config) {
+Router.prototype._loadModule = function(modDef, globals) {
 
     var self = this;
+
+    // First, check if we have a copy of this module in the cache, so that we
+    // can share it.
+    var cachedModule = self._modules.get(modDef);
+    if (cachedModule && cachedModule._parentGlobals === globals) {
```

Comment:
> The map keys equality are determined by the semantics of a === operator, so only references to the exact same object instance would be equal. 

This is a conservative default, which shouldn't reduce sharing opportunities in practice, as the globals object identity would likely differ in that case anyway.

> Yet another problem is with globals equality. === doesn't work as a deep equality on objects, so the second part of the condition would be almost always false.

This is on purpose, as complex objects like functions can't be compared otherwise. Again, I'd say this is mostly academic.